src/core/layertree/qgslayertreemodel.cpp

```diff
@@ -199,6 +199,10 @@ QVariant QgsLayerTreeModel::data( const QModelIndex &index, int role ) const
       {
         return QgsLayerItem::iconRaster();
       }
+      else if ( layer->type() == QgsMapLayer::MeshLayer )
```

Comment:
Could you change this block to a switch so we get warnings if new layer types are added in future?

Reply:
sure I can, but I am not sure I have time to add another layer type this year... I say, I layer type per decade is fine with me.

Reply:
@nyalldawson looking at the code the logic there is quite complex and not sure if I can refactor it to switch without change of behaviour for plugin layers. Could we postpone the refactoring after 3.6?